src/main/java/org/folio/marc4ld/service/ld2marc/resource/Resource2MarcRecordMapperImpl.java

```diff
@@ -121,6 +133,40 @@ private DataField getDataField(Ld2MarcFieldRuleApplier b2mRule, Resource resourc
     return field;
   }
 
+  private void addDateFields(Record marcRecord, Resource resource) {
+    addUpdatedDateField(marcRecord, resource);
+  }
+
+  private void addUpdatedDateField(Record marcRecord, Resource resource) {
+    if (resource.getTypes().equals(Set.of(INSTANCE))) {
+      var optionalWork = LdUtil.getWork(resource);
+      optionalWork.ifPresentOrElse(work -> {
+          var optionalDate = chooseDate(resource.getUpdatedAt(), work.getUpdatedAt());
+          optionalDate.ifPresent(date -> addDateField(marcRecord, date));
+        },
+        () -> ofNullable(resource.getUpdatedAt())
+          .ifPresent(date -> addDateField(marcRecord, date))
+      );
+    }
+  }
+
+  private Optional<Date> chooseDate(Date instanceDate, Date workDate) {
+    if (anyNull(instanceDate, workDate)) {
+      return Stream.of(instanceDate, workDate)
+        .filter(Objects::nonNull)
+        .findFirst();
+    }
+    return instanceDate.after(workDate) ? of(instanceDate) : of(workDate);
+  }
+
+  private void addDateField(Record record, Date date) {
```

Comment:
minor - you can rename this method to `addUpdateDateField`

Reply:
done.